ls1intum/ArTEMiS
# Bugfix/instructor status/use participation websocket for results

### Checklist

### Description


The instructor status does not update atm, because it has no subscription to the participation websocket service.
I added the subscription and updated the unit tests. There are a couple of new files in the spec folder, because I had to add mocks to test the component together with the websocket service.

### Steps for Testing


1. Log in to ArTEMiS
2. Go to programming exercise table / view
3. Execute a build run that changes the status of the template or solution repository (e.g. by creating a build error in the solution repository)
=> The warning should appear and disappear as expected (without reloading the page!).

### Screenshots


![image](https://user-images.githubusercontent.com/28230611/58797742-0c3e2780-8601-11e9-8ae4-8ecb0f91fe20.png)

![image](https://user-images.githubusercontent.com/28230611/58797758-1b24da00-8601-11e9-8a67-3f2f1b9e983e.png)
(Please ignore the build result of the solution repository, this is part of a another branch and doesn't affect this PR)
